### ISSUE TITLE
MaxChainedCallsOnSameLine false positive for deeply nested type

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#6218]
## Expected Behavior
A type `Nav.List.Params.Groups.Source.Profiles` shouldn't violate the `MaxChainedCallsOnSameLine`

## Observed Behavior
It does

## Steps to Reproduce
```kotlin
sealed class Nav {
  object List : Nav {
    sealed interface Params {
      object Groups : Params {
        enum class Source {
          Profiles
        }
      }
    }
  }
}
```

## Your Environment
* Version of detekt used: 1.23.0
